### DISCUSSION TITLE
PUT  metrics timeout

### DISCUSSION THREAD

**Discussion by yushengery** [prometheus/pushgateway#606] [Q&A]
I am using pushgateway to receive metrics pushed by the server through the Prometheus client. When querying through /metrics, I found that there are about 300,000 metrics on the Pushgateway. The size of data pushed by the client using the PUT API is around 80KB. However, I frequently encounter latency comes out to be more than 1 minute. What could be the reason for this?

**Comment by beorn7** [comment]
The Pushgateway is not designed for this amount of metrics. It's made for hundreds of metrics, not hundreds of thousands.

When you push metrics to the Pushgateway, it has to check that all the metrics together are still consistent after the push. With so many metrics already on your Pushgateway, this will take a long time.
